client/state/notices/middleware.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { truncate, forEach } from 'lodash';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { errorNotice } from 'state/notices/actions';
+import { POST_DELETE_FAILURE } from 'state/action-types';
+import { getSitePost } from 'state/posts/selectors';
+
+const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: [
```

Comment:
>For example a failed delete in a post list, versus a failed delete in the Calypso editor?

Not sure this is something we'd want to do in practice, though it should be do-able; you could, for example, switch on `state.ui.section`.